lib/doh.h

```diff
@@ -97,12 +105,44 @@ struct dohaddr {
   } ip;
 };
 
+#ifdef USE_HTTPSRR
+
+/*
+ * These are the code points for DNS wire format SvcParams as
+ * per draft-ietf-dnsop-svcb-https
+ * Not all are supported now, and even those that are may need
+ * more work in future to fully support the spec.
+ */
+#define HTTPS_RR_CODE_ALPN            0x01
+#define HTTPS_RR_CODE_NO_DEF_ALPN     0x02
+#define HTTPS_RR_CODE_PORT            0x03
+#define HTTPS_RR_CODE_IPV4            0x04
+#define HTTPS_RR_CODE_ECH             0x05
+#define HTTPS_RR_CODE_IPV6            0x06
+
+/*
+ * These may need escaping when found within an alpn string
+ * value.
+ */
+#define COMMA_CHAR                    ','
+#define BACKSLASH_CHAR                '\\'
+
+struct dohhttps_rr {
+  size_t len; /* raw encoded length */
```

Comment:
What is the maximum length of these RR fields? Maybe we can save a few bytes by using something smaller than `size_t` ?

Reply:
16  bits max as per [this](https://datatracker.ietf.org/doc/html/draft-ietf-dnsop-svcb-https#section-2.2) - I changed to ``uint16_t`` for the struct.

Is the concern to make the array of those structs smaller? If so, then the above is probably an ok change though those values get mapped to ``size_t`` in some API calls, which I'm guessing is ok?